ui/pages/send/send-content/send-content.container.js

```diff
@@ -24,6 +31,43 @@ function mapStateToProps(state) {
   const recipient = getRecipient(state);
   const recipientWarningAcknowledged =
     getRecipientWarningAcknowledgement(state);
+  const isBuyableChain = getIsBuyableChain(state);
+  const draftTransaction = getCurrentDraftTransaction(state);
+
+  const { currentCurrency, nativeCurrency, provider, unapprovedTxs } =
+    state.metamask;
+  const { chainId } = provider;
+  const editingTransaction = unapprovedTxs[draftTransaction.id];
+  const txData = {
+    txParams: {
+      gasPrice:
+        editingTransaction?.txParams?.gasPrice ??
+        draftTransaction.gas?.gasPrice,
+      gas: editingTransaction?.userEditedGasLimit
+        ? editingTransaction?.txParams?.gas
+        : draftTransaction.gas?.gasLimit,
+      maxFeePerGas:
+        editingTransaction?.txParams?.maxFeePerGas ??
+        draftTransaction.gas?.maxFeePerGas,
+      maxPriorityFeePerGas:
+        editingTransaction?.txParams?.maxPriorityFeePerGas ??
+        draftTransaction.gas?.maxPriorityFeePerGas,
+      value: draftTransaction.amount?.value,
+      type:
+        editingTransaction?.txParams?.type ?? draftTransaction.transactionType,
+    },
+    userFeeLevel: editingTransaction?.userFeeLevel,
+  };
+
+  const {
+    hexTransactionAmount,
+    hexMinimumTransactionFee,
+    hexMaximumTransactionFee,
+    hexTransactionTotal,
+  } = transactionFeeSelector(state, txData);
+
+  const { useNativeCurrencyAsPrimaryCurrency } = getPreferences(state);
```

Comment:
Rather than adding all of these to sent-content, just to pass them to the new gas-display component...couldn't we just use hooks in the gas-display component to get that information? Also, when a transaction is edited its details are plugged into the draftTransaction object and it becomes the current draft. You *shouldn't* have to use either or. Can you let me know if you're experiencing something different?

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/c64948a3c0494308f824cdc443a9b04b010b9054).